receiver/macosunifiedloggingreceiver/config.go

```diff
@@ -331,3 +333,60 @@ func hasBalancedParentheses(s string) bool {
 	}
 	return count == 0
 }
+
+// resolveArchivePath takes a path (potentially with glob pattern)
+// and returns a list of resolved archive directory paths
+func resolveArchivePath(pattern string) ([]string, error) {
+	var resolved []string
+	seen := make(map[string]bool)
+
+	// Clean the pattern
+	pattern = filepath.Clean(pattern)
+
+	// Check if pattern contains glob characters
+	hasGlob := strings.ContainsAny(pattern, "*?[]")
+
+	if hasGlob {
+		// Use doublestar for glob matching
+		matches, err := doublestar.FilepathGlob(pattern)
+		if err != nil {
+			return nil, fmt.Errorf("invalid glob pattern %q: %w", pattern, err)
+		}
+
+		// Validate each matched path
+		for _, match := range matches {
+			if err := validateArchivePath(match); err != nil {
+				// Skip invalid matches and continue
+				continue
+			}
+			// Deduplicate paths
+			if !seen[match] {
+				resolved = append(resolved, match)
+				seen[match] = true
+			}
+		}
```

Comment:
This isn't making a lot of sense to me why are we deduplicating the matches? Are we seeing doubles with doublestar somewhere?